.github/workflows/build.yml

```
@@ -0,0 +1,54 @@
+name: Build and Test
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main, beta ]
+  pull_request:
+    branches: [ main, beta ]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        node: [ 16, 18, 20 ]
+
+    name: Build Package (Node ${{ matrix.node }})
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --include=dev
```

Since `NODE_ENV` is never set anywhere the flag can likely be omitted.
```suggestion
        run: npm ci
```